shell/browser/api/electron_api_base_window.h

```diff
@@ -274,6 +275,8 @@ class BaseWindow : public gin_helper::TrackableObject<BaseWindow>,
 #endif
 
   v8::Global<v8::Value> content_view_;
+  // ids of attached browser views z-index sorted.
+  std::list<int32_t> browser_views_ids_by_z_indexes_;
   std::map<int32_t, v8::Global<v8::Value>> browser_views_;
```

Comment:
I'm good with the API, but I think we should implement this feature by putting the BrowserView values in a `std::vector` sorted by z-index.

Using `std::list` will be several orders of magnitude slower than using `std::vector`, because of CPU cache. Also using a ID map to store BrowserView was an overkill design, for small number of elements `std::map` is not necessarily faster than `std::vector`.

```suggestion
  std::vector<v8::Global<v8::Value>> browser_views_;
```

Reply:
Hey @zcbenz 
Thanks for having a look and your feedback!
I'll update this PR on weekends 👍